src/firebase/CRUDServices/userServices.js

```diff
@@ -21,97 +17,120 @@ export default class UserServices {
+        createUserWithEmailAndPassword(user.email, user.password);
+      return res;
+    } catch (error) {
+      const errorCode = error.code;
+      const errorMessage = error.message;
+      return { errorCode, errorMessage };
     }
   }
 
   /**
-   * Get all exsisting users.
-   * @return {User[]} array of user object.
+   * Send new user email verification when they register.
+   * @return {Response} result.
    */
-  getUsers = async () => {
+  sendEmailVerification = async () => {
     try {
-      const userRef = db.collection(self.users);
-      const snapshot = await userRef.get();
-      const res = [];
-      if (snapshot.empty) {
-        // TODO: add logic here
-        return [];
-      }
-
-      snapshot.forEach( (doc) => {
-        res.push(doc.data());
+      firebase.auth().onAuthStateChanged(async (user)=>{
+        if (user) {
+          const res = await user.sendEmailVerification();
+          return res;
+        } else {
+          return { errorCode: 'auth/no user',
+            errorMessage: 'did not find user' };
+        }
       });
-      return res;
-    } catch (err) {
-      // TODO: add logic here
+    } catch (error) {
+      const errorCode = error.code;
+      const errorMessage = error.message;
+      return { errorCode, errorMessage };
     }
   }
 
-
   /**
-   * Get user by specific id.
-   * @param {String} userId id of user.
-   * @return {User} user object.
+   *
+   * @param {User} user user object contains email and password.
+   * @return {Response} result.
    */
-  getUserById = async (userId) => {
+  signUp = async (user) => {
     try {
-      const userRef = db.collection(self.users)
-        .doc(userId);
-      const doc = await userRef.get();
-      if (!doc.exists) {
-        // TODO: add logic here
-        return {};
-      } else {
-        return doc.data();
+      const res = await this.postUser(user);
+      if (res) {
+        // this.sendEmailVerification();
+        return { status: 200, message: 'created' };
       }
-    } catch (err) {
-      // TODO: add logic here
+    } catch (error) {
+      const errorCode = error.code;
+      const errorMessage = error.message;
+      return { errorCode, errorMessage };
     }
   }
 
   /**
-   * Update specific attributs of user by id.
-   * @param {String} userId id of user.
-   * @param {User} user updating attributes and values.
-   * @return {Boolean} succeed or not.
+   * Sign in user.
+   * @param {User} user user object with email and password.
+   * @return {user} authenticated user object.
    */
-  updateUserById = async (userId, user) => {
+  signIn = async (user) => {
     try {
-      const userRef = db.collection(self.users)
-        .doc(userId);
-      await userRef.update(user);
-      return true;
-    } catch (err) {
-      // TODO: add logic here
-      return false;
+      const userCredential = await firebase.auth()
+        .signInWithEmailAndPassword(user.email, user.password);
```

Comment:
I thought we were doing usernames. Is it email instead?

Reply:
Yes, sorry for the confusion. Firestore seems like not supporting username as standard login.